exercises/grains/example.jl

```diff
@@ -0,0 +1,18 @@
+"""Calculate the number of grains on square `square`."""
+function on_square(square)
+    check_square_input(square)
+    2 ^ (square - 1)
+end
+
+"""Calculate the total number of grains after square `square`."""
+function total_after(square)
+    check_square_input(square)
+    sum(map(on_square, 1:square))
+end
+
+"""Validate an arbitrary square."""
+function check_square_input(square)
```

Comment:
You could leave both the 0.7 and the 0.6 version in and check which version is currently used. That'll save some time later on when we migrate to 0.7/1.0